src/Text/Pretty/Simple.hs

```diff
@@ -665,6 +687,10 @@ layoutStringAnsi opts = fmap convertStyle . layoutString opts
 --
 -- __Compactness options__
 --
+-- >>> pPrintOpt CheckColorTty defaultOutputOptionsDarkBg {outputOptionsCompact = True} $ B ( B ( B ( B A ) ) )
+-- B
+--     ( B ( B ( B A ) ) )
```

Comment:
While some existing behaviours have changed, the fact that this one doesn't with the main commit is important. I think it's fine that we insert more newlines, given that we can use the compact option to bring things back together to some extent if desired.